scipy/stats/tests/test_stats.py

```diff
@@ -2881,6 +2881,22 @@ def test_1samp_alternative(self):
         assert_allclose(p, self.P1_1_g)
         assert_allclose(t, self.T1_1)
 
+    def test_1samp_nan_alternative(self):
```

Comment:
These tests fit into the existing tests. I don't like the inconsistencies of the existing tests (e.g. some functions test 1d, some 2d, some 3d, and only ever on one axis), but fixing that is out of scope.

Reply:
I seems that the purpose of this test is to be an intuitive one. But why have an intuitive test for `ttest_1samp` and not the others?

Reply:
> These tests fit into the existing tests. I don't like the inconsistencies of the existing tests (e.g. some functions test 1d, some 2d, some 3d, and only ever on one axis), but fixing that is out of scope.

I agree. We can fix that in another PR. Do you think tracking this in an issue would be helpful?

> I seems that the purpose of this test is to be an intuitive one. But why have an intuitive test for `ttest_1samp` and not the others?

I might have added this by mistake. Didn't mean to add an intuitive test here. I didn't add intuitive tests because it has already been done while adding `alternative` parameter. Also, tests against R have been added there. So, I will remove this.